src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedHandler.cs

```diff
@@ -0,0 +1,60 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Core.Features.Security.Authorization;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Security;
+using Microsoft.Health.Dicom.Core.Messages.Retrieve;
+using Microsoft.Health.Dicom.Core.Web;
+
+namespace Microsoft.Health.Dicom.Core.Features.Retrieve;
+internal class RetrieveRenderedHandler : BaseHandler, IRequestHandler<RetrieveRenderedRequest, RetrieveRenderedResponse>
+{
+    private readonly IRetrieveRenderedService _retrieveRenderedService;
+
+    public RetrieveRenderedHandler(IAuthorizationService<DataActions> authorizationService, IRetrieveRenderedService retrieveRenderedService)
+        : base(authorizationService)
+    {
+        _retrieveRenderedService = EnsureArg.IsNotNull(retrieveRenderedService, nameof(retrieveRenderedService));
+    }
+
+    public async Task<RetrieveRenderedResponse> Handle(RetrieveRenderedRequest request, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(request, nameof(request));
+
+        if (await AuthorizationService.CheckAccess(DataActions.Read, cancellationToken) != DataActions.Read)
+        {
+            throw new UnauthorizedDicomActionException(DataActions.Read);
+        }
+
+        ValidateRetrieveRenderedRequest(request);
+
+        RetrieveRenderedResponse renderedResponse = await _retrieveRenderedService.RetrieveRenderedImageAsync(request, cancellationToken);
+
+        return renderedResponse;
+    }
+
+    private static void ValidateRetrieveRenderedRequest(RetrieveRenderedRequest request)
```

Comment:
Validation should happen in the Renderedservice. Hndler should just to Authorization check

Reply:
Both the handlers for retrieve instance and retrieve metadata do the validation in the handler and not in the service - I put it here to keep it consistent across our service. Should I move the validation for all 3 to the service or should I just leave them in the handler?